### DISCUSSION TITLE
Deploy custom client problem

### DISCUSSION THREAD

**Discussion by ilovebioz** [mozilla/hubs#5359] [Hubs Cloud Questions] [answered]
We followed the instruction to deploy our custom client to the hubs cloud but we got 2 problems:

1. after deploying the admin page missing, we can not access the admin page.
2. the new hub client refused to call external API, it worked normally local.

Do you have any experience with these?
![screenshot_1649815046](https://user-images.githubusercontent.com/28090157/163084391-17fcc313-b0b4-4c9e-bc2f-36a77bbb0f07.png)

**Comment by rawnsley** [comment]
Try this workaround in case it's the same problem https://github.com/mozilla/hubs/issues/4208#issuecomment-1021577510

**Comment by conath** [comment, marked as answer]
I had similar problems with a custom client including admin page broken sometimes, depending on CSP (Content-Security-Policy) settings. As the browser is telling you, the CSP is denying access to your URL. 

Usually you edit the CSP allow-list through the Hubs Admin page => Setup / Server Settings / Advanced. There are several "Extra Content-Security-Policy rules" fields that you can add any third party domains to (separated by spaces). For example, in my client I added these to the "Extra Content Security Policy connect-src Rules":

`https://firestore.googleapis.com https://identitytoolkit.googleapis.com https://fonts.gstatic.com https://securetoken.googleapis.com`

If the Admin page does not load due to CSP errors, you should be able to fix it by editing or emptying the field in AWS Systems Manager / Parameter store. The fields are called `/<path to your instance>/reticulum/extra_csp/<field>`, for example the above `connect-src` rule is in `/.../reticulum/extra_csp/connect_src` for me.
